site/README.md

```diff
@@ -19,10 +19,21 @@ limitations under the License.
 
 # Apache Calcite docs site
 
-This directory contains the code for the Apache Calcite web site,
-[calcite.apache.org](https://calcite.apache.org/).
-
-You can build the site manually using your environment or use the docker compose file.
+This directory contains the code and templates for generating the Apache Calcite website,
+[calcite.apache.org](https://calcite.apache.org/). The actual generated content of the website
+is present in [calcite-site](https://github.com/apache/calcite-site) repository.
+
+We want to deploy project changes (for example, new committers, PMC members or upcoming talks)
+immediately, but we want to deploy documentation of project features only when that feature appears
+in a release.
+
+The procedure for deploying the changes to the website is outlined below:
+1. Push the commit with the changes to the `master` branch of this repository.
+2. Cherry-pick the commit from the `master` branch to the `site` branch of this repository.
+3. Checkout the `site` branch and build the website either [manually](#manually) or using
+[docker compose](#using-docker) (preferred).
```

Comment:
I would write "docker compose" as either "`docker-compose`" the name of the CLI command or "Docker Compose", the natural language name of the tool.